rclpy/rclpy/node.py

```diff
@@ -999,6 +999,11 @@ def _set_parameters_atomically_common(
         elif self._on_set_parameters_callbacks:
             for callback in self._on_set_parameters_callbacks:
                 result = callback(parameter_list)
+                if not isinstance(result, SetParametersResult):
```

Comment:
_Random bystander here_
Python is not a strong typed language, so a callback is free to return anything.

So in the concept of duck typing, why not only check what you really need?

`if not hassatr(result, successful):`

Not feeling strong about it though, so feel free to ignore this :)

Reply:
i think right thing to do here is to check the expected object is returned by callback, instead of checking the attribution.